test.lua

```diff
@@ -866,4 +879,35 @@ describe("Flow", function()
             ]])
         end)
     end)
+
+    describe("Flow.embed", function ()
+        local embedded_form = flow.make_gui(function ()
+            return gui.Label{label = "This is the embedded form!"}
+        end)
+        test("supports nil prefix", function ()
+            local _e, _x, s = render_from_func_to_string(function (p, _x)
+                return gui.HBox{
+                    gui.Label{label = "asdft"},
+                    embedded_form:embed{ player = p },
+                    gui.Label{label = "ffaksksdf"},
+                }
+            end)
+            assert.equals(
```

Comment:
I'm concerned that this doesn't normalize the tree. Any ideas?

Reply:
Maybe just a function that parses the formspec, flattens it, and sets `formspec_version` to a constant value (so the tests don't break if flow's formspec version is bumped)?